src/main/java/org/apache/ibatis/type/TypeHandlerRegistry.java

```diff
@@ -393,9 +393,11 @@ private void register(Type javaType, JdbcType jdbcType, TypeHandler<?> handler)
       Map<JdbcType, TypeHandler<?>> map = typeHandlerMap.get(javaType);
       if (map == null || map == NULL_TYPE_HANDLER_MAP) {
         map = new HashMap<>();
+        map.put(jdbcType, handler);
         typeHandlerMap.put(javaType, map);
+      } else {
+        map.put(jdbcType, handler);
       }
```

Comment:
Wow...did this actually happen on production?

Regarding the change, how about updating `typeHandlerMap` at the end?
i.e.

```java
      Map<JdbcType, TypeHandler<?>> map = typeHandlerMap.get(javaType);
      if (map == null || map == NULL_TYPE_HANDLER_MAP) {
        map = new HashMap<>();
      }
      map.put(jdbcType, handler);
      typeHandlerMap.put(javaType, map);
```

Reply:
Yep, this happening a couple times in a week on startup. But our apps work like as "lamba" (many cold startups in a loop).

I agree, this will save a couple of code lines :+1:

Reply:
I see. That explains it.
The fix is merged and should be available in the latest 3.5.4-SNAPSHOT.
Please let me know if you noticed anything.

Thank you for the PR, @PVlyubinskiy !